Refonte de la documentation (avec Eleventy)

loup-brun opened the pull request:
Voir #845

loup-brun committed:
refonte doc: premier pas d’intégration eleventy

ggrossetie commented on docs/.gitignore:
```diff
@@ -100,5 +105,5 @@ sftp-config.json
 ### https://raw.github.com/github/gitignore/master/Global/NotepadPP.gitignore
 
-# Notepad++ backups #
-*.bak
```
(nitpick): tu peux ajouter `.idea/` en ignore et supprimer le dossier

loup-brun replied:
Oups pardon! J’ai oublié de le faire…

ggrossetie commented on docs/.eleventy.js:
```diff
@@ -0,0 +1,20 @@
+ * https://www.11ty.dev/docs/config/
+ */
+
+const pluginWebc = require("@11ty/eleventy-plugin-webc");
```
Je découvre webc, ça a l'air assez sympa et _simple_ à utiliser 😍

loup-brun replied:
En effet, j’aime beaucoup pouvoir composer avec des components (colocation, single-file-components…), le paradigme webc semble en super bonne voie d’intégration dans eleventy!

loup-brun committed:
gitignore: .idea

loup-brun committed:
docs: suite refonte avec eleventy

note: il y a un bug actuellement qui empêche les changements faits aux components d’être actualisés en mode de développement lorsqu’ils sont importés par un layout, voir cette issue sur github:
https://github.com/11ty/eleventy/issues/2903

Le fix sera livré dans la prochaine version patch d’Eleventy (donc bientôt)

loup-brun committed:
un peu de travail exploratoire webc

loup-brun committed:
edit: refonte site doc

wip #845

ggrossetie committed:
Add Netlify config

loup-brun committed:
travail sur disposition et style

tenter une première implémentation des nouvelles couleurs CRCÉN

loup-brun committed:
pnpm install

loup-brun committed:
avancer doc: corrections de style, disposition plus robuste

loup-brun committed:
page.webc => docs.webc

loup-brun committed:
tenir compte de date modif et version

loup-brun committed:
avancer sur disposition docs

loup-brun committed:
refonte doc: layout, page footer, alert-blocks

loup-brun committed:
doc: plus de composantes, ménage, documenter code

loup-brun committed:
layouts: fix comments

loup-brun committed:
Merge remote-tracking branch 'origin/master' into 845-refonte-documentation

loup-brun committed:
afficher la vraie date de dernière modification

voir https://www.11ty.dev/docs/dates/

loup-brun committed:
edit(doc) readme: documenter SSG eleventy, répertoire, étapes pour reproduire

loup-brun commented:
@antoinentl @ggrossetie @thom4parisot
Cette PR est prête pour révision. Il resterait à configurer l’hébergement (avec le compte Netlify cela devrait être très facile), mais aussi à entrer certaines métadonnées pour que les pages s’affichent correctement (ex. balise `title` au lieu d’un `h1` dans le corps).

TODO:
- [x] Entrer métadonnées @giuliaferretti12 @RochDLY **EDIT** c’est fait
- [x] Réviser les chemins vers les images (certains sont brisés) : `/uploads/images/<image.ext>`
- [x] Configurer l’hébergement et le déploiement EDIT: https://deploy-preview-858--stylo-docs.netlify.app/fr/ ✅
- [ ] Si le site n’est pas servi à la racine (`/`) mais plutôt à `/docs`, il faudra faire une autre ronde de révision.

loup-brun commented:
![stylo-doc-neue-menu](https://github.com/EcrituresNumeriques/stylo/assets/9596476/48439b05-73a3-4834-b73d-08602fd5559d)
![stylo-doc-neue](https://github.com/EcrituresNumeriques/stylo/assets/9596476/2c543221-ec6a-4083-8750-881eb181b157)

loup-brun committed:
edit(docs) readme

ggrossetie commented:
Si c'est possible je pense que ça pourrait être bien d'avoir https://docs.stylo-dev.huma-num.fr

ggrossetie commented on docs/.eleventy.js:
```diff
@@ -0,0 +1,37 @@
+
+  // Copier les contenus du dossier `uploads/` tels quels
+  // (sans le dossier `uploads/` dans le chemin de destination)
+  eleventyConfig.addPassthroughCopy("uploads");
```
D'après le déploiement sur Netlify le répertoire `uploads` est conservé :

https://stylo-docs.netlify.app/uploads/images/Login-Huma-Num-V2.PNG

D'ailleurs il faut revoir les liens vers les images afin d'utiliser un chemin absolu `/images`.

ggrossetie replied:
D'après la documentation il faut préciser la target: https://www.11ty.dev/docs/copy/#change-the-output-directory

loup-brun replied:
> D'après le déploiement sur Netlify le répertoire `uploads` est conservé

La documentation précédente ne nécessitait aucun build (affichage des fichiers markdown côté client avec MDwiki), c’était donc copié tel quel.

Pour des fins de clarté, je pense que c’est plus simple si le dossier `uploads/` est copié tel quel, et que les liens vers les images utilisent le chemin absolu `/uploads/images` (cela serait aussi compatible avec l’installation actuelle; les liens vers les images ne seront pas brisés lors du déploiement du nouveau site).

ggrossetie replied:
On peut conserver le répertoire `uploads` mais il faudra quand même changer le chemin vers les images car actuellement le chemin est relatif à la page.

loup-brun replied:
Oui, la plupart des pages ont un lien relatif `uploads/images` – j’ai mentionné cela par écrit aux autres (Roch, Giulia) qui vont se charger de l’édition de la doc!

ggrossetie replied:
OK parfait 👍

loup-brun committed:
docs: ne pas appliquer layout sur pages de redirection

loup-brun committed:
docs: date: Last Modified par défaut

loup-brun committed:
docs: enlever `date` dans frontmatter

Last Modified par défaut, voir 1d0cabd4

loup-brun committed:
docs: navbar menu pos absolute sur petits écrans

possibilité pour afficher le menu, mais ce n’est pas hyper beau

loup-brun committed:
docs: ajustements navigation

loup-brun committed:
docs: migrer métadonnées (titres frontmatter) et chemins images

loup-brun committed:
docs: gérer lignes dans templates

loup-brun committed:
docs: enrichir readme pour contribuer à la doc

loup-brun committed:
docs: règles horizontales dans prose

loup-brun committed:
docs: ajout composant <link-button>

et duh… remplacer chemins /uploads pour vrai

loup-brun committed:
docs: edit readme (documenter link-button)

loup-brun committed:
docs: prose-block: limiter largeur du contenu

loup-brun committed:
docs(layout): améliorer disposition générale + barre (surtout pour mobile)

loup-brun committed:
docs(style): micro détail top-bar

loup-brun committed:
docs(layout/style): fix prefers-color-scheme meta

loup-brun committed:
docs: cleanup test v3

loup-brun committed:
docs(theme): font Bitter et favicon (svg)

loup-brun committed:
docs(theme) fix theme-color

loup-brun committed:
docs(theme): supporter base url pour section

Pratique pour faire d’autres sections, ex. /v2 ex. /blog ex. /nouvelles

loup-brun committed:
docs: fix chemin section documentation

loup-brun commented:
## Thème/couleurs de la nouvelle doc

Est-ce que quelqu’un pourrait donner une rétroaction par rapport à l’intégration des nouvelles couleurs de la CRCÉN dans la doc? (cc en particulier @marviro, mais aussi les contributeurs comme @antoinentl @mattheyje @Mmellet) Voici un aperçu du travail sur la doc en l’état ([accessible en prévisualisation ici](https://deploy-preview-858--stylo-docs.netlify.app/fr/syntaxemarkdown/))&nbsp;:

![Screenshot 2023-06-13 at 11-38-51 Stylo - un éditeur de texte sémantique Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/4b8eddc1-964c-42ae-9bb2-6f181bd5b43b)

![Screenshot 2023-06-13 at 11-38-43 Stylo - un éditeur de texte sémantique Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/4a5fa7f0-e50d-4a46-8500-6535cb7f2550)

---

Il serait aussi possible d’avoir un jeu de couleurs complètement neutre, reflétant l’état actuel de l’interface de Stylo. Voici ce à quoi ça pourrait ressembler (jeu de couleurs en niveaux de gris)&nbsp;:

![Screenshot 2023-06-13 at 11-58-55 Syntaxe markdown Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/1c25a17e-8e4e-48af-9b4b-5043037d47a4)

![Screenshot 2023-06-13 at 11-59-00 Syntaxe markdown Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/bae6baf3-d909-4aba-aa0b-ad6cd312a071)

loup-brun committed:
doc(refonte): jeu de couleurs neutre et clair par défaut

loup-brun commented:
Styleguide: https://deploy-preview-858--stylo-docs.netlify.app/fr/styleguide/

loup-brun committed:
docs(layout): pageheader

loup-brun committed:
docs: commencer à intégrer search

loup-brun committed:
docs(refonte): travail sur module recherche

loup-brun committed:
docs: search en opération!

loup-brun commented:
Ajout de la fonctionnalité de recherche (tel que suggéré par @Mmellet).

![stylo-doc-recherche](https://github.com/EcrituresNumeriques/stylo/assets/9596476/25203752-3259-478d-b0b3-ffcf5be860cb)

loup-brun committed:
docs: disposition search

aussi corriger grosseur champ de recherche pour iOS
https://stackoverflow.com/questions/2989263/disable-auto-zoom-in-input-text-tag-safari-on-iphone

antoinentl commented:
Je suis partagé entre la version _couleur_ et la version en noir et blanc (ou niveaux de gris), les deux sont lisibles. Voici mes arguments :

- version _couleur_ : un peu de couleurs fait du bien, mais nous pourrions réinjecter ces couleurs dans Stylo pour les rares endroits où il y en a ;
- version noire et blanc : plus cohérente avec l'interface (actuelle) de Stylo, et peut-être plus élégante (mais c'est subjectif).

Je laisse @mattheyje @Mmellet et @marviro réagirent, le choix qui sera fait sera le bon choix.

ggrossetie commented on docs/src/en/about.md:
```diff
@@ -57,4 +58,4 @@ The researchers participating in the project are as follows:
 ## Developers
 
 - [Guillaume Grossetie](https://github.com/Mogztter)
```
```suggestion
- [Guillaume Grossetie](https://github.com/ggrossetie)
```

loup-brun replied:
Corrigé dans e8ad376.

mattheyje commented:
> Je suis partagé entre la version _couleur_ et la version en noir et blanc (ou niveaux de gris), les deux sont lisibles. Voici mes arguments :
> 
>     * version _couleur_ : un peu de couleurs fait du bien, mais nous pourrions réinjecter ces couleurs dans Stylo pour les rares endroits où il y en a ;
> 
>     * version noire et blanc : plus cohérente avec l'interface (actuelle) de Stylo, et peut-être plus élégante (mais c'est subjectif).
> 
> 
> Je laisse @mattheyje @Mmellet et @marviro réagir, le choix qui sera fait sera le bon choix.

Je suis d'accord avec tous les points soulevés par Antoine et vois aussi des avantages aux deux possibilités...peut-être que le NB est plus élégant en effet ? À voir avec @Mmellet et @marviro

antoinentl commented:
Après en avoir discuté avec @RochDLY et @marviro, on part sur la version en noir et blanc/neutre/niveaux de gris. Et nous pouvons déployer dès qu'on le peut, avec quelques tâches supplémentaires :

- [ ] rediriger le lien vers la documentation depuis Stylo
- [ ] rediriger stylo-doc.ecrituresnumeriques.ca vers stylo.huma-num.fr/doc en 301 @antoinentl 
- [ ] rediriger toutes les pages (en 301) ? @loup-brun je suis preneur de ton avis

loup-brun committed:
style(docs): espacement pied de page

loup-brun committed:
style(docs): couleurs neutres pour style clair

ajout variable CSS --couleur-marque (ainsi que --anticouleur-marque)

loup-brun committed:
docs: résoudre conflit maj master

loup-brun committed:
doc: déplacer nouvau fichier (provenant de conflit préc.)

loup-brun committed:
docs: remplacer nom utilisateur github ggrossetie

d’après https://github.com/EcrituresNumeriques/stylo/pull/858#discussion_r1245451869

loup-brun committed:
maj couleurs favicon

loup-brun committed:
docs: fix styles alert-block

loup-brun committed:
docs: fix couleurs inline-search.webc

loup-brun committed:
docs: fix couleurs main-navigation.webc

loup-brun committed:
docs: variables.css

loup-brun committed:
docs: toucher stylefuide (pour structure cliquable)

loup-brun commented:
màj du 8 septembre :

- l'interface de la doc est en niveaux de gris.
- les fichiers sont à jour avec `master`.
- favicon avec couleurs neutres (différence système clair et foncé).
- certains agencements de couleurs ont été corrigés.

---

![Screenshot 2023-09-08 at 10-50-47 Stylo - un éditeur de texte sémantique Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/bcbe94ab-b01f-40a1-b325-41222a7655d2)

---

![Screenshot 2023-09-08 at 10-51-03 Stylo - un éditeur de texte sémantique Stylo](https://github.com/EcrituresNumeriques/stylo/assets/9596476/35250ffa-6513-4c8a-a2d7-18a20594dab4)
